src/hera/workflows/io/_io_mixins.py

```diff
@@ -64,8 +64,11 @@ def _get_parameters(cls, object_override: Optional[Self] = None) -> List[Paramet
 
         for field, field_info in get_fields(cls).items():
             if get_origin(annotations[field]) is Annotated:
-                param = get_args(annotations[field])[1]
+                # Copy so as to not modify the Input fields themselves
+                param = get_args(annotations[field])[1].copy()
```

Comment:
This copy is inspired by [this one](https://github.com/argoproj-labs/hera/blob/main/src/hera/workflows/script.py#L506). However, I can also make sense of it by looking at [this test](https://github.com/argoproj-labs/hera/blob/main/tests/script_annotations/pydantic_io_v1.py#L68). In this case, we would not want to modify the annotations of the `ParamOnlyInput` class, because it may be used by another script template (possibly with different defaults).